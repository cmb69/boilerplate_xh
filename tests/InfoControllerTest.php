<?php

namespace Boilerplate;

use ApprovalTests\Approvals;
use PHPUnit\Framework\MockObject\Stub;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    /** @var TextBlocks&Stub */
    private $textBlocks;

    /** @var View */
    private $view;

    /** @var InfoController */
    private $sut;

    public function setUp(): void
    {
        $this->textBlocks = $this->createStub(TextBlocks::class);
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["boilerplate"]);
        $this->sut = new InfoController(
            "./plugins/boilerplate/",
            $this->textBlocks,
            new FakeSystemChecker(),
            $this->view
        );
    }

    public function testRenderInfo(): void
    {
        $this->textBlocks->method("getDataFolder")->willReturn("./content/boilerplate/");
        Approvals::verifyHtml(($this->sut)()->output());
    }
}
