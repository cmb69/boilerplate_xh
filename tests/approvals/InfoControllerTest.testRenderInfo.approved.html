<!-- Boilerplate info -->
<h1>Boilerplate 2.1-dev</h1>
<h4>System check</h4>
<div><p class="xh_fail">PHP version ≥ 7.1.0</p>
</div>
<div><p class="xh_warning">Folder “./plugins/boilerplate/css“ writable</p>
</div>
<div><p class="xh_warning">Folder “./plugins/boilerplate/languages/“ writable</p>
</div>
<div><p class="xh_warning">Folder “./content/boilerplate/“ writable</p>
</div>
